pkg/astnormalization/remove_scalar_duplicates.go

```diff
@@ -0,0 +1,71 @@
+package astnormalization
+
+import (
+	"github.com/jensneuse/graphql-go-tools/pkg/ast"
+	"github.com/jensneuse/graphql-go-tools/pkg/astvisitor"
+)
+
+func removeScalarDuplicates(walker *astvisitor.Walker) {
+	visitor := removeScalarDuplicatesVisitor{
+		Walker: walker,
+	}
+	walker.RegisterScalarTypeDefinitionVisitor(&visitor)
```

Comment:
You can remove `walker.RegisterScalarTypeDefinitionVisitor(&visitor)` and `LeaveScalarTypeDefinition` function. Because you already call `walker.RegisterEnterScalarTypeDefinitionVisitor`.